test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/FhirStorageTests.cs

```diff
@@ -597,93 +604,321 @@ public async Task GivenATransactionHandler_WhenATransactionFailsFailedRequest_Th
         }
 
         [Fact]
-        [FhirStorageTestsFixtureArgumentSets(DataStore.CosmosDb)]
-        public async Task GivenAnUpdatedResource_WhenUpdateSearchIndexForResourceAsync_ThenResourceGetsUpdated()
+        public async Task GivenAnUpdatedResource_WhenUpdatingSearchParameterIndexAsync_ThenResourceMetadataIsUnchanged()
```

Comment:
@feordin I updated this test so that it can be used for both Cosmos and SQL, and, as a result, I had to remove the check that the Cosmos ETag changed.

I think this should be alright, since we check that a reindexed resource is returned in searches with the new search parameter in `ReindexJobTests`. I could be missing something though, so please let me know if this looks alright!